qemu/tests/qemu_guest_agent.py

```diff
@@ -111,12 +111,29 @@ def _cleanup_open_session(self):
     @error_context.context_aware
     def _check_ga_pkg(self, session, cmd_check_pkg):
         '''
-        Check if the package is installed
+        Check if the package is installed, for rhel8 need to check
+        if the current pkg is the latest one.
+
         :param session: use for sending cmd
         :param cmd_check_pkg: cmd to check if ga pkg is installed
         '''
-        error_context.context("Check whether qemu-ga is installed.", logging.info)
+        error_context.context("Check whether qemu-ga is installed.",
+                              logging.info)
         s, o = session.cmd_status_output(cmd_check_pkg)
+        if s == 0 and self.params["os_variant"] == 'rhel8':
+            # qemu-guest-agent-2.12.0-88.module+el8.1.0+4233+bc44be3f.x86_64
+            error_context.context("Check if the installed pkg is the latest"
+                                  " one for rhel8 guest.", logging.info)
+            version_list = []
+            build_latest = re.sub(r'/', '-', self.qga_pkg_latest_url)
+            for pkg in [o, build_latest]:
+                pattern = r"guest-agent-(\d+.\d+.\d+-\d+).module"
+                qga_v = re.findall(pattern, pkg, re.I)[0]
+                version_list.append(qga_v)
+            logging.info("The installed and the latest pkg version is"
+                         " %s" % version_list)
+            if version_list[1] != version_list[0]:
```

Comment:
@xiagao I think if version_list[1] != version_list[0],  you should install version_list[1].    shouldn't return False.

Reply:
return False means it need to install the right pkg.